src/bin/wit-bindgen.rs

```diff
@@ -1,12 +1,21 @@
 use anyhow::{Context, Result};
 use clap::Parser;
+use lazy_static::lazy_static;
 use std::path::PathBuf;
 use wit_bindgen_core::{wit_parser, Files, Generator};
 use wit_parser::Interface;
 
+lazy_static! {
+    pub static ref VERSION: String = build_info();
+}
+
+/// Helper for passing VERSION to opt.
+fn version() -> &'static str {
+    &VERSION
+}
```

Comment:
This is a bit of an iteration beyond what Cargo does but I think this entire `VERSION` string can be built in the build script? Could that be done to avoid needing `lazy_static!` and building it at runtime here?

Additionally I don't think this build will work when `.git` is missing or the `git` executable itself because then the env vars aren't set by the build script.

Reply:
This is really a good point!

I build the enriched version information in `build.rs` and thus got rid of `lazy_static!`. For the second comment, thanks for pointing out a error handling case! What I did is to use `CARGO_PKG_VERSION` as the version info if `.git` is missing or the `git` executable itself is missing.